se_core/test/geometry/octree_collision_unittest.cpp

```diff
@@ -79,80 +80,300 @@ class OctreeCollisionTest : public ::testing::Test {
-  const Eigen::Vector3i width = {2, 2, 2};
+  const Eigen::Vector3i width = {3, 3, 3};
+
+  //sphere
+  const Eigen::Vector3i center = {61, 13, 253};
+  const int radius = 1;
 
   /* Update leaves as occupied node */
-  se::VoxelBlock<testT> * block = oct_.fetch(56, 12, 254);
+  se::VoxelBlock<testT> *block = oct_.fetch(56, 12, 254);
   const Eigen::Vector3i blockCoord = block->coordinates();
-  int x, y, z, blockSide; 
+  int x, y, z, blockSide;
   blockSide = (int) se::VoxelBlock<testT>::side;
+//  testing::internal::CaptureStdout();
   int xlast = blockCoord(0) + blockSide;
   int ylast = blockCoord(1) + blockSide;
   int zlast = blockCoord(2) + blockSide;
-  for(z = blockCoord(2); z < zlast; ++z){
-    for (y = blockCoord(1); y < ylast; ++y){
-      for (x = blockCoord(0); x < xlast; ++x){
-        if(x < xlast/2 && y < ylast/2 && z < zlast/2)
+  for (z = blockCoord(2); z < zlast; ++z) {
+    for (y = blockCoord(1); y < ylast; ++y) {
+      for (x = blockCoord(0); x < xlast; ++x) {
+        if ((blockCoord(0)) < x && x < (xlast - 4) && (blockCoord(1)) < y && y < (ylast - 4)
+            && (blockCoord(2)) < z && z < (zlast - 4)) {
           block->data(Eigen::Vector3i(x, y, z), 2.f);
-        else
+        } else {
           block->data(Eigen::Vector3i(x, y, z), 10.f);
-
+        }
       }
     }
   }
 
-  const collision_status collides = collides_with(oct_, test_bbox, width, 
-      test_voxel);
+  const collision_status collides = collides_with(oct_, test_bbox, width, test_voxel);
+  const collision_status collidesSphere = isSphereCollisionFree(oct_, center, radius);
+//  std::string output = testing::internal::GetCapturedStdout();
+//  EXPECT_TRUE(false) << output;
   ASSERT_EQ(collides, collision_status::empty);
+  ASSERT_EQ(collidesSphere, collision_status::empty);
+
+}
+
+TEST_F(OctreeCollisionTest, CollisionLeaf) {
+  // Allocated block: {56, 8, 248};
+  const Eigen::Vector3i test_bbox = {57, 9, 249};
+  const Eigen::Vector3i width = {5, 5, 5};
+
+  // sphere
+  const Eigen::Vector3i center = {60, 11, 251};
+  const int radius = 2;
+  /* Update leaves as occupied node */
+  se::VoxelBlock<testT> *block = oct_.fetch(56, 12, 254);
+  const Eigen::Vector3i blockCoord = block->coordinates();
+  int x, y, z, blockSide;
+  blockSide = (int) se::VoxelBlock<testT>::side;
+  testing::internal::CaptureStdout();
+  int xlast = blockCoord(0) + blockSide;
+  int ylast = blockCoord(1) + blockSide;
+  int zlast = blockCoord(2) + blockSide;
+  for (z = blockCoord(2); z < zlast; ++z) {
+    for (y = blockCoord(1); y < ylast; ++y) {
+      for (x = blockCoord(0); x < xlast; ++x) {
+        if ((blockCoord(0) + 1) < x && x < (xlast - 2) && (blockCoord(1) + 1) < y && y < (ylast - 2)
+            && (blockCoord(2) + 1) < z && z < (zlast - 2)) {
+          block->data(Eigen::Vector3i(x, y, z), 2.f);
+        } else {
+          block->data(Eigen::Vector3i(x, y, z), 10.f);
+        }
+      }
+    }
+  }
```

Comment:
The code above is duplicated in roughly half of the tests. It should be in the SetUp() function instead creating a second octree instance (`oct_collision_`?) which can then be accessed by the individual tests.